tests/third_party/cupy/linalg_tests/test_decomposition.py

```diff
@@ -0,0 +1,248 @@
+import unittest
+
+import numpy
+import pytest
+
+import dpnp as cupy
+from tests.helper import has_support_aspect64
+from tests.third_party.cupy import testing
+from tests.third_party.cupy.testing import condition
+
+
+def stacked_identity(xp, batch_shape, n, dtype):
+    shape = batch_shape + (n, n)
+    idx = xp.arange(n)
+    x = xp.zeros(shape, dtype=dtype)
+    x[..., idx, idx] = 1
+    return x
+
+
+@testing.parameterize(
+    *testing.product(
+        {
+            "full_matrices": [True, False],
+        }
+    )
+)
+# @testing.fix_random()
```

Comment:
I think we can uncomment that. To do this you need:
- in file `tests/third_party/cupy/testing/__init__.py` uncomment
```python
from tests.third_party.cupy.testing.random import generate_seed
```
- update `tests/third_party/cupy/testing/random.py` file with:
```python
def do_setup(deterministic=True):
    ...
    _old_cupy_random_states = cupy.random.dpnp_iface_random._dpnp_random_states
    cupy.random.dpnp_iface_random._dpnp_random_states = {}
    ...
    assert (
        cupy.random.dpnp_iface_random._dpnp_random_states
        is not _old_cupy_random_states
    )
   ...

def do_teardown():
    ...
    cupy.random.dpnp_iface_random._dpnp_random_states = _old_cupy_random_states
    ...
```